framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2632,56 +2643,56 @@ protected void checkMethodInvocability(
         }
     }
 
+    /**
+     * Check that the (explicit) annotations on a new class tree are comparable to the result type
+     * of the constructor. Issue an error if not.
+     *
+     * <p>Issue a warning if the annotations on the constructor invocation is a subtype of the
+     * constructor result type. This is equivalent to down-casting.
+     *
+     * @param invocation
+     * @param constructor
+     * @param newClassTree
+     * @return
+     */
     protected boolean checkConstructorInvocation(
```

Comment:
Should we just make this method consistent with the others and make it return `void`? The single caller doesn't check the returned value and most other `checkXXX` methods are void.

Reply:
Ping on this.

Reply:
Sure, but that's a breaking change, so we would need to increase the minor number for this release.  We are only at 2.7.0, so I think we should wait.

Reply:
This PR makes many breaking changes, so increasing the minor version seems fine to me.

Reply:
This PR does not change any public or protect methods, so all external checker will still compile.  Though I did changed the semantics of AnnotatedTypeFactory#newClassTree, so that counts as breaking.  Either way, this is low priority.

Reply:
It adds a few protected methods and if type systems don't override them they will get unexpected warnings. So at least calling out these changes in the changelog seems important.

Reply:
Yes, I absolutely need to add a change log for this PR.  I think the only new warning is the warning for any constructor result that isn't top.  The other warnings were in the last release.

Reply:
FYI, there are breaking changes in other pull requests, so I changed this to a void method.